Fix MPI backend PG initialization

H-Huang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#92847 Fix MPI backend PG initialization**

Fixes #92573

Add test to check that all default backends can be initialized to prevent the above from regressing in the future.

H-Huang committed:
Fix MPI backend PG initialization

[ghstack-poisoned]

awgu commented on test/distributed/test_c10d_common.py:
```diff
@@ -1532,6 +1532,34 @@ def tearDown(self):
 
+    def test_init_process_group_for_all_backends(self):
+        for backend in dist.Backend.backend_list:
+            # skip is the backend is not available on the system
```
nit:
```suggestion
            # skip if the backend is not available on the system
```

H-Huang committed:
Update test/distributed/test_c10d_common.py

Co-authored-by: Andrew Gu <31054793+awgu@users.noreply.github.com>

H-Huang commented:
@pytorchbot merge

H-Huang commented:
@pytorchbot merge

H-Huang committed:
Update on "Fix MPI backend PG initialization"


Fixes #92573

Add test to check that all default backends can be initialized to prevent the above from regressing in the future.

[ghstack-poisoned]

H-Huang commented:
@pytorchbot merge
